### PR TITLE
RPRBLND-1841: add External Data options support

### PR DESCRIPTION
**Pull request by DagerD**
### PURPOSE
Need to add support for the External Data option support

### EFFECT OF CHANGE

- Changed UI for file selection for nodes in texture2d and texture3d node categories
- Extended Data now works (tested with images)

### TECHNICAL STEPS

- Reworked MaterialX classes builder. Now image selector is PointerProperty
- Changed UI for file selection in nodes
- Added cache_image_file method for cache images in temp dir.

### NOTES FOR REVIEWERS
You need to rebuild MaterialX classes to see the changes.

### CODE REVIEW

File: src/hdusd/mx_nodes/node_tree.py [hunk 1]
```diff
@@ -178,3 +180,28 @@ def update_(self):
         for material in bpy.data.materials:
             if material.hdusd.mx_node_tree and material.hdusd.mx_node_tree.name == self.name:
                 material.hdusd.update()
+
+        ng = bpy.data.node_groups.get(self.name, None)
```
Comment (bnagirniak) [inline src/hdusd/mx_nodes/node_tree.py:184]: This whole code is not needed here

**Review by bnagirniak — CHANGES_REQUESTED** [review]
MxNode.compute() should be adjusted.

File: src/hdusd/engine/viewport_engine.py [hunk 2]
```diff
@@ -351,7 +351,10 @@ def material_update(cls, material):
     def update_material(self, mat):
         stage = self.cached_stage()
         material.sync_update_all(stage.GetPseudoRoot(), mat)
-        self.render_engine.tag_redraw()
+        try:
+            self.render_engine.tag_redraw()
```
Comment (bnagirniak) [inline src/hdusd/engine/viewport_engine.py:355]: Probably error is happen earlier, please check.

**Review by VascoPi — APPROVED** [review]
Tested, no errors, works well.

**Review by bnagirniak — APPROVED** [review]
Tested - works good now.

**Pull request merged**